source/binding/Bitstream.cpp

```diff
@@ -679,9 +822,178 @@ ConstantValue Bitstream::evaluateTarget(const StreamingConcatenationExpression&
 
     bitwidth_t bitOffset = 0;
     auto iter = packed.cbegin();
-    if (!unpackConcatenation(lhs, iter, packed.cend(), bitOffset, dynamicSize, context))
+    auto iterEnd = packed.cend();
+    if (!unpackConcatenation(lhs, iter, iterEnd, bitOffset, dynamicSize, context))
         return nullptr;
+
+    if (iter == iterEnd) {
+        ASSERT(dynamicSize == 0);
+        if (bitOffset > 0) {
+            // Target longer than source
+            context.addDiag(diag::BadStreamSize, lhs.sourceRange) << srcSize + bitOffset << srcSize;
+        }
+    }
+    else if (rhs.kind == ExpressionKind::Streaming) {
+        // Target shorter than source
+        auto tSize = srcSize - (*iter++)->bitstreamWidth() + bitOffset;
```

Comment:
Is there any chance this size computation can underflow and wrap around?

Reply:
No, this should never happen. "srcSize" is the sum of bitstreamWidth() from the entire vector, and "(*iter++)" is only an element of the vector. I add an assertion to ensure this assumption in the future.
```c++
        ASSERT(srcSize >= (*iter++)->bitstreamWidth());
```